src/OrderingUtil.js

```diff
@@ -0,0 +1,38 @@
+import OrderedMsgChain from './OrderedMsgChain'
+
+export default class OrderingUtil {
+    constructor(streamId, streamPartition, inOrderHandler, gapHandler, gapFillTimeout) {
+        this.streamId = streamId
+        this.streamPartition = streamPartition
+        this.inOrderHandler = inOrderHandler
+        this.gapHandler = gapHandler
+        this.gapFillTimeout = gapFillTimeout
+        this.orderedChains = {}
+    }
+
+    add(unorderedStreamMessage) {
+        const chain = this._getChain(unorderedStreamMessage.getPublisherId(), unorderedStreamMessage.getMsgChainId())
+        chain.add(unorderedStreamMessage)
+    }
+
+    _getChain(publisherId, msgChainId) {
+        const key = publisherId + msgChainId
+        if (!this.orderedChains[key]) {
+            this.orderedChains[key] = new OrderedMsgChain(publisherId, msgChainId, this.inOrderHandler, this.gapHandler, this.gapFillTimeout)
+        }
+        return this.orderedChains[key]
+    }
+
+    addError(err) {
```

Comment:
Same comment as before in `OrdererMsgChain`. No need for this class to be tied to the structure of the error.